ext/standard/math.c

```diff
@@ -875,11 +877,39 @@ PHPAPI int _php_math_basetozval(zval *arg, int base, zval *ret)
 			c -= 'A' - 10;
 		else if (c >= 'a' && c <= 'z')
 			c -= 'a' - 10;
-		else
+		else if (isspace(c)) {
+			if (num == 0 && fnum == 0) {
+				continue;
+			}
+			lastchar = 1;
 			continue;
-
-		if (c >= base)
+		} else {
+			invalidchars++;
 			continue;
+		}
+
+		if (lastchar == 1 && c > 0 && c < base) {
+			invalidchars++;
+		}
+		lastchar = 0;
+
+
+		if (c >= base) {
+			if (*(s-2) == '0' && num == 0) {
+				if (base == 16 && c == 33) { /* allow the second char to be x */
```

Comment:
```suggestion
				if (base == 16 && c == 'x') { /* allow the second char to be x */
```

and similar for the others. I think we should be allowing both `0x` and `0X` here, as we also do so in integer literals.